packages/vendure-plugin-invoices/src/credit-invoice-generation.process.ts

```diff
@@ -0,0 +1,29 @@
+import { OrderProcess } from '@vendure/core';
+import { InvoiceService } from './services/invoice.service';
+
+let invoiceService: InvoiceService;
+
+export const creditInvoiceGenerationProcess: OrderProcess<'Cancelled'> = {
+  init(injector) {
+    invoiceService = injector.get(InvoiceService);
+  },
+
+  async onTransitionStart(_, toState, data) {
+    const previousInvoiceForOrder =
```

Comment:
We might need to account for the situation where an order doesn't have a previous invoice. I imagine this:

1. When a previous invoice isnt there, we log an error but dont throw
2. When invoice generation fails its good to throw an error, because this should block cancellation